### PR TITLE
[RHOL-631] finish justification follows validation

### PR DESCRIPTION
**Pull request by AbnerZheng**
## Overview
it should be that for a block, we have `toLatestMessages(block.justifications).keys == bonds(creatorJustification(block)).keys` where .keys return all the validators.

### Which JIRA issue does this PR relate to? If there is not a JIRA issue addressing this work, please create one now and add the link here.
https://rchain.atlassian.net/browse/RHOL-631

### Complete this checklist before you submit the PR
- [X] This PR contains no more than 200 lines of code, excluding test code.
- [X] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [X] You assigned one person to review this PR

### CODE REVIEW

**Comment by AbnerZheng** [comment]
@KentShikama PTAL.

File: casper/src/main/scala/coop/rchain/casper/Validate.scala [hunk 1]
```diff
@@ -397,6 +399,36 @@ object Validate {
...
+   * This check must come before Validate.parents
+   */
+  def justificationFollows[F[_]: Monad: Log: BlockStore](
+      b: BlockMessage,
+      genesis: BlockMessage,
+      dag: BlockDag): F[Either[InvalidBlock, ValidBlock]] =
+    for {
+      latestMessages <- ProtoUtil.toLatestMessage[F](b.justifications, dag)
+      validators     = latestMessages.keySet
+      creatorLatestBlock <- ProtoUtil.creatorJustification(b).foldM(genesis) {
```
Comment (KentShikama) [inline casper/src/main/scala/coop/rchain/casper/Validate.scala]: Please refactor the ProtoUtil.creatorJustification: we now store the latest messages in memory so you shouldn't have to fetch the creator justification.

File: casper/src/main/scala/coop/rchain/casper/Validate.scala [hunk 2]
```diff
@@ -397,6 +399,36 @@ object Validate {
...
+      creatorBonds = ProtoUtil.bonds(creatorLatestBlock).map(_.validator).toSet
+      result       = creatorBonds == validators
+      status <- if (result) {
+                 Applicative[F].pure(Right(Valid))
+               } else {
+                 for {
+                   _ <- Log[F].warn(
+                         ignore(
+                           b,
+                           "the justifications of block are not follow from bonds of creator justification block"))
```
Comment (KentShikama) [inline casper/src/main/scala/coop/rchain/casper/Validate.scala]: the justification of the block does not follow from ... 

Also can you mkstring from the list of justifications and print them out with PrettyPrinter.buildString(...) ?

File: casper/src/main/scala/coop/rchain/casper/Validate.scala (hunk 1 above)
Comment (AbnerZheng) [reply casper/src/main/scala/coop/rchain/casper/Validate.scala]: Done

File: casper/src/main/scala/coop/rchain/casper/Validate.scala (hunk 2 above)
Comment (AbnerZheng) [reply casper/src/main/scala/coop/rchain/casper/Validate.scala]: Done

**Pull request merged**